tests/layer_tests/tensorflow_tests/test_tf_Gather.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestGather(CommonTFLayerTest):
+
+    def create_indices_constant(self):
+        pass
+
+    def create_gather_net(self, data_shape, indices, axis, batch_dims, **kwargs):
+        import tensorflow as tf
+
+        tf.compat.v1.reset_default_graph()
+
+        with tf.compat.v1.Session() as sess:
+            data = tf.compat.v1.placeholder(tf.float32, data_shape, 'data')
+            indices = tf.constant(indices, dtype=tf.int32)
+            gather = tf.gather(data, indices, axis=axis, batch_dims=batch_dims, name='gather_output')
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
```

Comment:
The IR will be pretty simple, so please, add a reference net code.

Reply:
I this it should be nit. A lot of other layer tests do not have `ref_net`

https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_Concat.py#L61
https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_Select.py#L51
https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_Eltwise.py#L53

on the other hand for onnx gather there are `concat` nodes for some reason and I don't why. https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/onnx_tests/test_gather.py#L189-L213

So for me it looks not so simple and we have inference results which are more confident confirmation that everything works fine. Can we skip `ref_net` for this time?

Reply:
postponed to implement further in a separate PR